### ISSUE TITLE
RAR7093 error in Glassfish 3.1.2

### ISSUE THREAD

**Issue by THatanaka** [eclipse-ee4j/glassfish#22797]
## Environment Details
* GlassFish Version (and build number): 3.1.2 Full Platform
* JDK version: 7 update 80
* OS: Windows 10 Pro (Anniversary Update)
* Database: PostgreSQL 10.3

----------

Since I use Google translation, please forgive the insults in English.
Can I post here about GF 3?

In Glassfish 3, "RAR7093" error occurs irregularly.
The stack trace at the time of occurrence is as follows.

> org.postgresql.util.PSQLException: 同じPooledConnectionが開かれたので新しい接続は自動的に閉じられました。または、PooledConnectionは既に閉じられています。("The new connection was closed automatically as the same PooledConnection was opened. Or, PooledConnection is already closed.")
>	at org.postgresql.ds.PGPooledConnection$ConnectionHandler.invoke(PGPooledConnection.java:312)
>	at com.sun.proxy.$Proxy123.prepareStatement(Unknown Source)
>	at com.sun.gjc.spi.base.ConnectionHolder.prepareStatement(ConnectionHolder.java:562)
>	at com.sun.gjc.spi.jdbc40.ConnectionWrapper40.prepareCachedStatement(ConnectionWrapper40.java:255)
>	at com.sun.gjc.spi.jdbc40.ConnectionWrapper40.prepareCachedStatement(ConnectionWrapper40.java:52)
>	at com.sun.gjc.spi.ManagedConnection.prepareCachedStatement(ManagedConnection.java:993)
>	at com.sun.gjc.spi.jdbc40.ConnectionWrapper40.prepareStatement(ConnectionWrapper40.java:173)
>	at (my program - "Connection.PrepareStatement")

> RAR7093 : Error while cleaning up ManagedConnection
> java.lang.NullPointerException
> 	at com.sun.gjc.spi.ManagedConnection.getActualConnection(ManagedConnection.java:717)
>	at com.sun.gjc.spi.ManagedConnectionFactory.resetIsolation(ManagedConnectionFactory.java:513)
>	at com.sun.gjc.spi.ManagedConnection.resetConnectionProperties(ManagedConnection.java:496)
>	at com.sun.gjc.spi.ManagedConnection.cleanup(ManagedConnection.java:348)
>	at com.sun.enterprise.resource.allocator.AbstractConnectorAllocator.cleanup(AbstractConnectorAllocator.java:166)
>	at com.sun.enterprise.resource.pool.ConnectionPool.cleanupResource(ConnectionPool.java:1082)
>	at com.sun.enterprise.resource.pool.AssocWithThreadResourcePool.freeUnenlistedResource(AssocWithThreadResourcePool.java:262)
>	at com.sun.enterprise.resource.pool.ConnectionPool.resourceClosed(ConnectionPool.java:1017)
>	at com.sun.enterprise.resource.pool.PoolManagerImpl.putbackResourceToPool(PoolManagerImpl.java:428)
>	at com.sun.enterprise.resource.pool.PoolManagerImpl.resourceClosed(PoolManagerImpl.java:382)
>	at com.sun.enterprise.resource.listener.LocalTxConnectionEventListener.connectionClosed(LocalTxConnectionEventListener.java:77)
>	at com.sun.gjc.spi.ManagedConnection.connectionClosed(ManagedConnection.java:784)
>	at com.sun.gjc.spi.base.ConnectionHolder.close(ConnectionHolder.java:217)
>	at com.sun.gjc.spi.jdbc40.ConnectionHolder40.close(ConnectionHolder40.java:587)
>	at (my program - "Connection.close()" )

The conditions under which the error occurs are unknown and occur irregularly during operation.
Is this something I misconfigured or coded?

**Comment by bshannon** [comment]
Sorry, GlassFish 3 is very old.  Unless you can reproduce the problem with GlassFish 5
this bug report will be closed.

**Comment by THatanaka** [comment]
I'm sorry. We understand. I will test with Glassfish5.
As an additional question on that, are there any reports that an RAR7093 error has occurred in GF5?
Since the problem is occurring in the system delivered to customers, in order to replace GF5, we will replace one more in addition to the statement that "GF3 is old software that has not been supported yet." I want a basis for it.
For example, this error is solved in GF5, etc.

**Comment by bshannon** [comment]
Sorry, don't know.

Are you using the free version of GlassFish 3, or are you paying for support for the Oracle version of GlassFish 3?  I assume the former, so you can understand why there is no free support for a free product that is 10 years old.

**Comment by THatanaka** [comment]
I'm sorry.
I have been using GF3 (free version) for a long time and I encountered this problem for the first time recently.
I checked whether there was a similar problem, but I could not find any clues other than the words "This is an internal error" in the Oracle manual.
For this reason, I just wanted a clue to the cause and posted it here.

In any case, try retesting with GF5. Thank you very much.
